### PR TITLE
[new release] electrod (0.2.2)

### PR DESCRIPTION
**Pull request by grayswandyr**
Formal analysis for the Electrod formal pivot language

- Project page: <a href="https://grayswandyr.github.io/electrod">https://grayswandyr.github.io/electrod</a>
- Documentation: <a href="https://grayswandyr.github.io/electrod/api">https://grayswandyr.github.io/electrod/api</a>

##### CHANGES:

- switch to Iter and Stdlib
- reformat code
- fix bug related to update of Fmt

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Your package fails to build with OCaml 4.04:
```
#=== ERROR while compiling electrod.0.2.2 =====================================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/grayswandyr/electrod/releases/download/0.2.2/electrod-0.2.2.tbz)
# path                 ~/.opam/4.04/.opam-switch/build/electrod.0.2.2
# command              ~/.opam/4.04/bin/dune build -p electrod -j 2
# exit-code            1
# env-file             ~/.opam/log/electrod-15-ca0711.env
# output-file          ~/.opam/log/electrod-15-ca0711.out
### output ###
#       ocamlc src/.libelectrod.objs/byte/libelectrod__Fmtc.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.libelectrod.objs/byte -I /home/opam/.opam/4.04/lib/base -I /home/opam/.opam/4.04/lib/base/caml -I /home/opam/.opam/4.04/lib/base/shadow_stdlib -I /home/opam/.opam/4.04/lib/bytes -I /home/opam/.opam/4.04/lib/containers -I /home/opam/.opam/4.04/lib/containers/data -I /home/opam/.opam/4.04/lib/containers/monomorphic -I /home/opam/.opam/4.04/lib/containers/unix -I /home/opam/.opam/4.04/lib/fmt -I /home/opam/.opam/4.04/lib/gen -I /home/opam/.opam/4.04/lib/hashcons -I /home/opam/.opam/4.04/lib/iter -I /home/opam/.opam/4.04/lib/logs -I /home/opam/.opam/4.04/lib/mtime -I /home/opam/.opam/4.04/lib/mtime/os -I /home/opam/.opam/4.04/lib/ppx_deriving/runtime -I /home/opam/.opam/4.04/lib/ppx_inline_test/config -I /home/opam/.opam/4.04/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.04/lib/printbox -I /home/opam/.opam/4.04/lib/result -I /home/opam/.opam/4.04/lib/sexplib0 -I /home/opam/.opam/4.04/lib/stdcompat -I /home/opam/.opam/4.04/lib/stdlib-shims -I /home/opam/.opam/4.04/lib/uchar -I /home/opam/.opam/4.04/lib/visitors -intf-suffix .ml -no-alias-deps -open Libelectrod -o src/.libelectrod.objs/byte/libelectrod__Fmtc.cmo -c -impl src/Fmtc.pp.ml)
# File "src/Fmtc.ml", line 1:
# Error: The implementation src/Fmtc.pp.ml
#        does not match the interface src/.libelectrod.objs/byte/libelectrod__Fmtc.cmi:
#        Values do not match:
#          val ( **@ ) : 'a t -> 'b t -> ('a * 'b) t
#        is not included in
#          val ( **@ ) : 'a t -> 'a t -> 'a t
#        File "src/Fmtc.ml", line 98, characters 4-11: Actual declaration
#     ocamlopt src/.libelectrod.objs/native/libelectrod__Fmtc.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlopt.opt -w -40 -g -O3 -I src/.libelectrod.objs/byte -I src/.libelectrod.objs/native -I /home/opam/.opam/4.04/lib/base -I /home/opam/.opam/4.04/lib/base/caml -I /home/opam/.opam/4.04/lib/base/shadow_stdlib -I /home/opam/.opam/4.04/lib/bytes -I /home/opam/.opam/4.04/lib/containers -I /home/opam/.opam/4.04/lib/containers/data -I /home/opam/.opam/4.04/lib/containers/monomorphic -I /home/opam/.opam/4.04/lib/containers/unix -I /home/opam/.opam/4.04/lib/fmt -I /home/opam/.opam/4.04/lib/gen -I /home/opam/.opam/4.04/lib/hashcons -I /home/opam/.opam/4.04/lib/iter -I /home/opam/.opam/4.04/lib/logs -I /home/opam/.opam/4.04/lib/mtime -I /home/opam/.opam/4.04/lib/mtime/os -I /home/opam/.opam/4.04/lib/ppx_deriving/runtime -I /home/opam/.opam/4.04/lib/ppx_inline_test/config -I /home/opam/.opam/4.04/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.04/lib/printbox -I /home/opam/.opam/4.04/lib/result -I /home/opam/.opam/4.04/lib/sexplib0 -I /home/opam/.opam/4.04/lib/stdcompat -I /home/opam/.opam/4.04/lib/stdlib-shims -I /home/opam/.opam/4.04/lib/uchar -I /home/opam/.opam/4.04/lib/visitors -intf-suffix .ml -no-alias-deps -open Libelectrod -o src/.libelectrod.objs/native/libelectrod__Fmtc.cmx -c -impl src/Fmtc.pp.ml)
# File "src/Fmtc.ml", line 1:
# Error: The implementation src/Fmtc.pp.ml
#        does not match the interface src/.libelectrod.objs/byte/libelectrod__Fmtc.cmi:
#        Values do not match:
#          val ( **@ ) : 'a t -> 'b t -> ('a * 'b) t
#        is not included in
#          val ( **@ ) : 'a t -> 'a t -> 'a t
#        File "src/Fmtc.ml", line 98, characters 4-11: Actual declaration
```

File: packages/electrod/electrod.0.2.2/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+license: "MPL-2.0"
+dev-repo: "git+https://github.com/grayswandyr/electrod.git"
+doc: "https://grayswandyr.github.io/electrod/api"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "4.04"}
+  "dune" {build & >= "1.4"}
```
Comment (kit-ty-kate) [inline packages/electrod/electrod.0.2.2/opam]:
```suggestion
  "dune" {>= "1.4"}
```
Dune shouldn't be tagged as a build dependency (see #14266)

File: packages/electrod/electrod.0.2.2/opam [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "4.04"}
+  "dune" {>= "1.4"}
+  "cmdliner" 
+  "containers" {>= "2.0"}
+  "fmt"
```
Comment (kit-ty-kate) [inline packages/electrod/electrod.0.2.2/opam]:
```suggestion
  "fmt" {>= "0.8.7"}
```
I'm guessing the error I showed you above comes from this missing constraint (just guessing, please double-check)

**Comment by grayswandyr** [comment]
I'm gonna try other changes first, so I'm closing this PR.
Thanks.

**Pull request closed without merging**